### ISSUE TITLE
Add caching to QuotaManager to allow to obtain quotas set to containers without lagging

### ISSUE THREAD

**Issue by Dilshat** [subutai-io/base#1681]
Currently each quota is read separately from subos. Thus to obtain all quotas for a single container, a considerable amount of time can be taken which would cause client to block. If we are requesting quotas for several container, this could become a blocker. To solve this we need to have a cache in Quota manager module that is regularly populated by a background job.

**Comment by akarasulu** [comment]
Why are we not tracking quotas from the beginning with each container creation, and modification?

**Comment by Dilshat** [comment]
we are setting standard quotas , and store only sizes: small medium ...
we don't store each separate quota value (like ram, cpu etc.)
also if quota is manually modified on subos, there will be a desync

**Comment by Dilshat** [comment]
Done

**Comment by akarasulu** [comment]
@Dilshat @soffokl 

This is a really bad design shortfall. Subutai CLI actions need to be announced to SS to prevent synchronization problems in the SS db. We really need to make sure this is done right. Please file another issue and let's fix this. Terrible undefined things can happen when the environment state and SS are not consistent.

**Comment by soffokl** [comment]
If you are planning constantly request quota binding to get information about all container it will be even more expensive then collecting this information in Agent. Where is the source requirement for this? I think we need to think about it more, before starting to implement it.

**Comment by Dilshat** [comment]
ok lets think more, meanwhile the current implementation will be tested, it is easy to change

**Comment by Dilshat** [comment]
Here a new issue - https://github.com/subutai-io/base/issues/1685
